### ISSUE TITLE
gautomach relion conflicts

### ISSUE THREAD

**Issue by rmarabini** [I2PC/scipion#1638]
Hi @pconesa, I include an issue reported by J. Benito.

If run in GUI, gautomatch seems to conflict with relion. That is, start gautomatch and rellion_refine  in parallel using the same GPU, gautomatch, for many micrographs will report 0 particles found.

The ideal solution to this problem would be to force these two protocols to tun in different GPUs. Since I think this is too hard to implement an alternative would be to check how many particles have been found by gautomatch. Gautomatch almost always finds particles so if 10% of the micrographs report 0 particles  very likely something went wrong. So if  10% of the micrographs report 0 a warninig should be issued 

        

[Reported by J Benito]

**Comment by delarosatrevin** [comment]
I think this is a bit out of scope of Scipion...this could open an entire Pandora's box. 
The users should be responsible for using a given GPU for some jobs and another one for other. 
Or if it is in the GUI, you should not run both at the same time if this conflicts happens. 
Do you think the conflicts are in the temporary files we generate in Scipion? In case case, what is the point of running both at the same time? 

@rmarabini , explain more if I haven't understood the point.

**Comment by rmarabini** [comment]
Hi @delarosatrevin,

Purpose: when acquiring data you may want to create a reconstruction from a small subset of the data set. This is an option that is getting quite popular here.

Solution Obviously we cannot create a queue system for GPUs but the solution proposed is a easy one. Just check that there is some output. 

I think, but I do not know, that the problem is the shared use of a GPU and that somebody. I do not know if Gautomatch o scipion is not processing properly  an error message.

**Comment by delarosatrevin** [comment]
I understand the purpose, but not quite well the proposed solution. I'm fine as long as the general mechanism are not twisted and the code become impossible to maintain.

**Comment by pconesa** [comment]
So, capturing properly an exception in Gautomatch to inform about an error.....is that what the idea?

**Comment by rmarabini** [comment]
I do not know.

It would be a good idea to check if Gautomatch return an error code or not
when the process does not finish. Regarding the exception caching this is
something only Gautomatch developers may do so it is out of our scope.

**Comment by josuegbl** [comment]
If I understand the problem, I guess what is happening is a wrong use of the GPUs, and, as @delarosatrevin said: is a litle bit out of the scope of the Scipion. I think the best solution is to force both GAUTOMATCH and Relion uses different GPUs (e.g. GAUTO uses GPU 0 and Relion GPU 1). Of Course, as Relion reserve all GPU memory for itself, any GPU program (gctf, gautomatch, etc) must not use these GPU!.
